mito-ai/mito_ai/completion_handlers/smart_debug_handler.py

```diff
@@ -1,66 +1,169 @@
 # Copyright (c) Saga Inc.
 # Distributed under the terms of the GNU Affero General Public License v3.0 License.
 
+from typing import List, Union, Optional, AsyncGenerator, Callable
+
 from openai.types.chat import ChatCompletionMessageParam
-from mito_ai.models import SmartDebugMetadata, MessageType
-from mito_ai.prompt_builders.smart_debug_prompt import create_error_prompt, remove_inner_thoughts_from_message
+from mito_ai.models import (
+    SmartDebugMetadata,
+    MessageType,
+    CompletionRequest,
+    CompletionStreamChunk,
+    CompletionReply,
+)
+from mito_ai.prompt_builders.smart_debug_prompt import (
+    create_error_prompt,
+    remove_inner_thoughts_from_message,
+)
 from mito_ai.providers import OpenAIProvider
 from mito_ai.message_history import GlobalMessageHistory
 from mito_ai.completion_handlers.completion_handler import CompletionHandler
 from mito_ai.completion_handlers.open_ai_models import MESSAGE_TYPE_TO_MODEL
 from mito_ai.completion_handlers.utils import append_chat_system_message
-__all__ = ["get_smart_debug_completion"]
+
+__all__ = ["get_smart_debug_completion", "stream_smart_debug_completion"]
+
 
 class SmartDebugHandler(CompletionHandler[SmartDebugMetadata]):
     """Handler for smart debug completions."""
-    
+
     @staticmethod
     async def get_completion(
```

Comment:
will be nice when we can delete this whole function for the functions we want to have streaming!